chapter_linear-networks/linear-regression-scratch.md

```diff
@@ -278,15 +278,13 @@ def sgd(params, grads, lr, batch_size):  #@save
 ```{.python .input}
 #@tab paddle
 #@save
-def sgd(params, lr, batch_size):
+def sgd(params, lr, batch_size):  
     """小批量随机梯度下降"""
-    revise_params=[]
     with paddle.no_grad():
-        for param in params:
-            param -= lr * param.grad / batch_size
-            param.stop_gradient = False
-            revise_params.append(param)
-        return revise_params
+        for i,param in enumerate(params):
+            param -= lr * params[i].grad/ batch_size
+            params[i].set_value(param.numpy())
```

Comment:
Is this the right way? What if the params are on a GPU device. In that case will `.numpy()` work? Even if it works, this means we'll be copying data to the CPU from GPU which is not right.

Reply:
We should probably use something like `param.data` here

Reply:
There is no param.data API for PaddlePaddle, but we can change Paddle Tensor's value by using set_value API， I find out the set_value API can receive the Tensor, so the sgd can be:

```
def sgd(params, lr, batch_size):  #@save
    """小批量随机梯度下降"""
    with paddle.no_grad():
        for i, param in enumerate(params):
            param -= lr * params[i].grad / batch_size
            params[i].set_value(param)
            params[i].clear_gradient()
```
is this ok?

Reply:
Yes, if this works should be much better.